js/ui/codeView.js

```diff
@@ -96,8 +98,16 @@ const _ensureHackDataFile = (function () {
             return keyfile;
 
         try {
-            keyfile.load_from_dirs('hack-data.ini', searchPaths,
+            const [, path] = keyfile.load_from_dirs('hack-data.ini', searchPaths,
                 GLib.KeyFileFlags.NONE);
+            monitorFile = Gio.file_new_for_path(path);
+            monitor = monitorFile.monitor_file(Gio.FileMonitorFlags.WATCH_HARD_LINKS, null);
```

Comment:
Have you tested that this actually works when you update the Flatpak? The full path you are monitoring is:

```
/var/lib/flatpak/app/com.endlessm.HackComponents/current/active/files/share/hack-components/hack-data.ini
                                                 ^^^^^^^ ^^^^^^
```

This file will never be overwritten in place, nor replaced in the same directory. The components indicated with `^^^` are symbolic links. When a new version of the app is installed, a new deployment is created and the symbolic link(s) updated.

Does GFileMonitor monitor all parent directories of a file?

Reply:
Yes, I've tested downgrading the flatpak app, launching the lightspeed and upgrading the HackComponents again and the change is detected, after the update I see the FtH button in the lightspeed app. The monitor is detecting this particular change in the file.

Reply:
See below. Both parts have been resolved.